src/CMakeLists.txt

```diff
@@ -33,8 +35,22 @@ if(CMAKE_EXTRA_GENERATOR OR MSVC_IDE)
     list(APPEND CUKE_SOURCES ${CUKE_HEADERS})
 endif()
 
-add_library(cucumber-cpp-nomain STATIC ${CUKE_SOURCES})
-add_library(cucumber-cpp STATIC ${CUKE_SOURCES} main.cpp)
+if(CUKE_ENABLE_SHARED_LIB)
+    set(CUKE_LINKAGE SHARED)
+else()
+    set(CUKE_LINKAGE STATIC)
+endif()
+
+add_library(cucumber-cpp-nomain ${CUKE_LINKAGE} ${CUKE_SOURCES})
+add_library(cucumber-cpp ${CUKE_LINKAGE} ${CUKE_SOURCES} main.cpp)
+
+set_target_properties(cucumber-cpp cucumber-cpp-nomain PROPERTIES
+    DEFINE_SYMBOL cucumber_cpp_EXPORTS
+)
+generate_export_header(cucumber-cpp)
```

Comment:
What about using the `EXPORT_FILE_NAME` parameter of `GENERATE_EXPORT_HEADER` (if I read correctly) to match the current file naming conventions for include files? `#include "cucumber-cpp_export.h"` does not read well